### PR TITLE
SignOutContext needs AuthenticationProperties

### PR DESCRIPTION
**Pull request by Praburaj**
Fixes https://github.com/aspnet/HttpAbstractions/issues/152

@Tratcher @HaoK

### CODE REVIEW

**Comment by Tratcher** [comment]
@brentschmaltz This is for OIDC & WsFed.

File: src/Microsoft.AspNet.Http.Core/DefaultHttpResponse.cs [hunk 1]
```diff
@@ -162,11 +160,11 @@ public override void SignIn(string authenticationScheme, [NotNull] ClaimsPrincip
             }
         }
 
-        public override void SignOut(string authenticationScheme)
+        public override void SignOut(string authenticationScheme, AuthenticationProperties properties)
         {
             var handler = HttpAuthenticationFeature.Handler;
 
-            var signOutContext = new SignOutContext(authenticationScheme);
+            var signOutContext = new SignOutContext(authenticationScheme, properties == null ? null : properties.Dictionary);
```
Comment (PinpointTownes) [inline src/Microsoft.AspNet.Http.Core/DefaultHttpResponse.cs]: `properties?.Dictionary`?

Comment (Praburaj) [inline src/Microsoft.AspNet.Http.Core/DefaultHttpResponse.cs]: sure. fixed.

**Comment by davidfowl** [comment]
We don't do optional parameters anymore.

**Comment by Praburaj** [comment]
@davidfowl Removed optional parameters on this method.

**Comment by Praburaj** [comment]
Updated PR. 

/CC : @muratg

**Comment by muratg** [comment]
Changes look good to me, I'd ping @brentschmaltz to see if he sees any problems, otherwise :shipit:

**Comment by Praburaj** [comment]
@brentschmaltz?

**Comment by brentschmaltz** [comment]
Adding some more context after speaking with Praburaj. there can be a lot of other parameters for signout: id_token, session_state, sid, idp, state etc.  I just saw the spec on Thursday afternoon and am digesting it. We may want to have first class properties similar to how OIDCMessage and WsFedMessage work. It may mean <templates> or derived types. Not sure. Signout it not as simple as we think.

**Comment by Praburaj** [comment]
Merged https://github.com/aspnet/HttpAbstractions/commit/58c45cd379774cf1fd683a12465cb52094b7193d

**Pull request closed without merging**